lib/Core/TerriaError.ts

```diff
@@ -19,32 +20,77 @@ function resolveI18n(i: I18nTranslateString | string) {
   return typeof i === "string" ? i : i18next.t(i.key, i.parameters);
 }
 
+/** `TerriaErrorSeverity` can be `Error` or `Warning`.
+ * Errors with severity `Error` are presented to the user. `Warning` will just be printed to console.
+ */
+export enum TerriaErrorSeverity {
+  /** Errors which should be shown to the user. This is the default value for all errors.
+   */
+  Error,
+  /** Errors which can be ignored by the user. These will be printed to console s
+   * For example:
+   * - Failing to load models (from share links or stories) if they are **NOT** in the workbench
+   */
+  Warning
+}
+
 /** Object used to create a TerriaError */
 export interface TerriaErrorOptions {
   /**  A detailed message describing the error.  This message may be HTML and it should be sanitized before display to the user. */
   message: string | I18nTranslateString;
+
   /** A short title describing the error. */
   title?: string | I18nTranslateString;
 
   /** The object that raised the error. */
   sender?: unknown;
+
   /** True if the user has seen this error; otherwise, false. */
   raisedToUser?: boolean;
 
   /** Error which this error was created from. This means TerriaErrors can be represented as a tree of errors - and therefore a stacktrace can be generated */
   originalError?: TerriaError | Error | (TerriaError | Error)[];
 
-  /** If true, lib\ReactViews\Notification\terriaErrorNotification.tsx will be used to display error message.
-   * If false, a plain old `Notification` will be used
+  /** TerriaErrorSeverity - will default to `Error`
+   * A function can be used here, which will be resolved when the error is raised to user.
    */
-  useTerriaErrorNotification?: boolean;
+  severity?: TerriaErrorSeverity | (() => TerriaErrorSeverity);
```

Comment:
does this need to be a function? The only time i can see it being used as a function is inside a TerriaError.combine which does the severity escalation anyway.

Reply:
It is needed here

https://github.com/TerriaJS/terriajs/blob/30d20887a731e452ace320658f692baa12033bb4/lib/Models/Terria.ts#L1172-L1187

As `this.workbench` may not be set before the `TerriaError` is created. This is due to how loading `initSources` work - and how you could have members added to the workbench array from multiple `initSources`